### PR TITLE
Added new tests for the merchant order refund flow

### LINKED ISSUE #28356: Write new e2e test: Merchant / Orders / Refund Order
Write a new e2e test that covers `Merchant / Orders / Refund Order` flow.

### PR DESCRIPTION
**Pull request by zhongruige**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds a new suite of e2e tests covering the `Merchant / Orders / Refund Order` flow, testing refunding an order. The run time is around ~3-4s for the test.

This PR also updates the `createSimpleOrder()` helper to return the order (post) ID.

Additionally, the following helper methods were added:
- `goToOrder( orderId )`: go to view a single order
- `updateOrderStatus( orderId, status )`: updates the status of the provided order ID
- `addProductToOrder( orderId, productName )` component which adds the provided productName to the passed in orderId

Closes https://github.com/woocommerce/woocommerce/issues/28356.

### How to test the changes in this Pull Request:

1. Make sure the new e2e tests pass locally and on Travis. The tests can be ran following [these instructions](https://github.com/woocommerce/woocommerce/tree/master/tests/e2e#running-tests).

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
N/A

### CODE REVIEW

**Comment by zhongruige** [comment]
Just saw the build failed for the test:

```
✕ can issue a refund by quantity (1431ms)
3023
3024  ● WooCommerce Orders > Refund an order › can issue a refund by quantity
3025
3026    TimeoutError: Element .do-manual-refund (text: "Refund £9.99 manually") not found
```

I’ll take a look into why the test is passing locally but failing in the pipeline.

**Comment by zhongruige** [comment]
> Just saw the build failed for the test:
> 
> ```
> ✕ can issue a refund by quantity (1431ms)
> 3023
> 3024  ● WooCommerce Orders > Refund an order › can issue a refund by quantity
> 3025
> 3026    TimeoutError: Element .do-manual-refund (text: "Refund £9.99 manually") not found
> ```
> 
> I’ll take a look into why the test is passing locally but failing in the pipeline.

Turns out it was the currency symbol; so I swapped swapped it from `£` => `$` in the tests.

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-refund.test.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		beforeAll(async () => {
+			await StoreOwnerFlow.login();
+			await createSimpleProduct();
+			orderId = await createSimpleOrder();
+			await addProductToOrder(orderId, simpleProductName);
+
+			// Get the currency symbol for the store's selected currency
+			await page.waitForSelector('.woocommerce-Price-currencySymbol')
+			let currencyElement = await page.$('.woocommerce-Price-currencySymbol')
+			currencySymbol = await page.evaluate(el => el.textContent, currencyElement)
```
Comment (rrennick) [inline tests/e2e/core-tests/specs/merchant/wp-admin-order-refund.test.js:33]: The above 3 lines are missing trailing `;`

Comment (zhongruige) [reply tests/e2e/core-tests/specs/merchant/wp-admin-order-refund.test.js]: Ack, good catch. Fixed!

**Comment by rrennick** [comment]
@zhongruige Nice work. Tested great for me.

@tammullen Leaving final review for you.

**Comment by zhongruige** [comment]
> @zhongruige Nice work. Tested great for me.

Thanks for the feedback!

**Review by tammullen — COMMENTED** [review]
Looks great, and they run great for me in headless.
I'm having some issues with running them in non-headless mode. The filter tests are failing when I'm doing this, seems like not all the orders for the filter tests aren't being created correctly. I'll keep looking into this but I don't think it will be a blocker.

**Comment by zhongruige** [comment]
> Looks great, and they run great for me in headless.
> I'm having some issues with running them in non-headless mode. The filter tests are failing when I'm doing this, seems like not all the orders for the filter tests aren't being created correctly. I'll keep looking into this but I don't think it will be a blocker.

Weird.. that got me curious so I just ran the filter tests too in non-headless and saw the same behavior. I removed the timeout override in the `beforeAll({});` setup on the filter tests and those worked fine in non-headless and (perhaps somewhat ironically) they failed in headless mode. I am curious if you're seeing the same behavior there, too.

**Review by tammullen — APPROVED** [review]
Approving as they run fine headless in Travis and locally. We can investigate the issues with non-headless separately.

**Comment by tammullen** [comment]
> I just ran the filter tests too in non-headless and saw the same behavior.

Thanks for checking and confirming!

> I am curious if you're seeing the same behavior there, too.

@zhongruige yes, I just tried this and got the same! We can investigate it separately from this PR.

**Comment by zhongruige** [comment]
> > I just ran the filter tests too in non-headless and saw the same behavior.
> 
> Thanks for checking and confirming!
> 
> > I am curious if you're seeing the same behavior there, too.
> 
> @zhongruige yes, I just tried this and got the same! We can investigate it separately from this PR.

Sounds good! I'll log an issue for it and put the details of what we're seeing into the description.

**Pull request merged**